apps/password-hasher/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use argon2::{
+    password_hash::{rand_core::OsRng, PasswordHash, PasswordHasher, PasswordVerifier, SaltString},
+    Argon2, Params,
+};
+
+use spin_sdk::http::{IntoResponse, Request, Response};
+use spin_sdk::http_component;
+
+#[http_component]
+fn handle_request(req: Request) -> anyhow::Result<impl IntoResponse> {
+    let mut password: String = "boson42".to_string();
+    let mut iterations: u32 = 5;
+    let mut mem_kib: u32 = 1000;
+
+    let parsed_url = url::Url::parse(req.uri()).unwrap();
+    let hash_query: std::collections::HashMap<_, _> =
+        parsed_url.query_pairs().into_owned().collect();
+
+    for (key, value) in hash_query.into_iter() {
+        match &*key {
+            // sleep=<ms> param simulates generic i/o time
+            "sleep" => {
```

Comment:
The sleep toggle originates from the beginning of this app which was derived from https://github.com/fermyon/spin/blame/13a133fded7c48fa5d1931207619379e737e9376/crates/trigger-http/benches/spin-http-benchmark/src/lib.rs.  Is this too kitchen-sinky?  Do we want a separate sleep app?

Reply:
Being kitchen-sinky seems like an _advantage_ to me for this sort of thing. You could maybe call this "benchmarking-load" or something to make that more clear?

Reply:
Oh I would maybe split these into two routes though, e.g. `/sleep?ms=...` and `/hash?cpu=...&mem=...` or something?

Reply:
I see the value of a kitchen sink app and just a sleep one. We could start with this as is, and if we want an app/endpoint that is only I/O, we can add it.